### PR TITLE
reparo: refine integration test

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What problem does this PR solve? 
reparo's integration test is too simple, should refine it
cherry pick https://github.com/pingcap/tidb-binlog/pull/621 to release 2.1

### What is changed and how it works?
have too many difference, so I just copy code

### Check List 

Tests 

 - Integration test

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -26,8 +26,9 @@ require (
 	github.com/onsi/ginkgo v1.8.0 // indirect
 	github.com/onsi/gomega v1.5.0 // indirect
 	github.com/pierrec/lz4 v2.0.5+incompatible // indirect
-	github.com/pingcap/check v0.0.0-20171206051426-1c287c953996
+	github.com/pingcap/check v0.0.0-20190102082844-67f458068fc8
 	github.com/pingcap/errors v0.11.0
+	github.com/pingcap/log v0.0.0-20190307075452-bd41d9273596
```
Comment (july2993) [inline go.mod]: don't use this for 2.1 ?

**Comment by kennytm** [comment]
@WangXiangUSTC Could you summarize what this PR changed for end-users (if this needs to be part of the release note)?

**Comment by WangXiangUSTC** [comment]
@kennytm I think don't need add to the release note, this pr add test and fix a bug
